pkg/cmd/node/node_list_components.go

```diff
@@ -0,0 +1,75 @@
+package node
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/dcos/dcos-cli/api"
+	"github.com/dcos/dcos-core-cli/pkg/health"
+	"github.com/dcos/dcos-core-cli/pkg/pluginutil"
+	"github.com/spf13/cobra"
+)
+
+func newCmdNodeListComponents(ctx api.Context) *cobra.Command {
+	var jsonOutput bool
+	var leader bool
+	var mesosID string
+	cmd := &cobra.Command{
+		Use:   "list-components",
+		Short: "Print a list of available DC/OS components on specified node",
+		Args:  cobra.NoArgs,
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			if !leader && mesosID == "" {
+				return fmt.Errorf("'--leader' or '--mesos-id' must be provided")
+			} else if leader && mesosID != "" {
+				return fmt.Errorf("unable to use leader and mesos id at the same time")
+			}
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ip := ""
+			if leader {
+				leader, err := mesosClient().Leader()
+				if err != nil {
+					return err
+				}
+				if leader.IP == "" {
+					return fmt.Errorf("invalid leader response, missing field 'ip'")
+				}
+				ip = leader.IP
+			} else {
+				agents, err := mesosClient().Agents()
+				if err != nil {
+					return err
+				}
+				for _, agent := range agents {
+					if mesosID == agent.AgentInfo.GetHostname() {
```

Comment:
As discussed, comparing against the IP/hostname is a bug, the `--mesos-id` option description specifies it refers to the node ID.